src/Payload.php

```diff
@@ -60,6 +60,41 @@ public function getClaims()
     }
 
     /**
+     * Checks if a payload matches some expected values.
+     *
+     * @param  array  $values
+     * @param  bool  $strict
+     *
+     * @return bool
+     */
+    public function matches(array $values, $strict = false)
+    {
+        if (empty($values)) {
+            return false;
+        }
+
+        foreach ($values as $key => $value) {
+            if (($strict && $this->get($key) !== $value) || $this->get($key) != $value) {
```

Comment:
@tymondesigns Let me know about this new approach added by https://github.com/tymondesigns/jwt-auth/pull/744/commits/6b6d52b856061e31fd6a6a9331f4c15710233fd5 ?